cpp/include/cudf/rolling.hpp

```diff
@@ -275,5 +275,133 @@ std::unique_ptr<column> rolling_window(
   std::unique_ptr<aggregation> const& agg,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief  Applies an offset-based window function (like LEAD/LAG) on specified column.
+ *
+ * Offset-based window functions include LEAD and LAG:
+ *   1. LEAD(N): Returns the row from the input column, at the specified offset past the
+ *      current row. If the offset crosses the grouping boundary or column boundary for
+ *      a given row, a "default" value is returned if available. The "default" value is
+ *      null, by default.
+ *   2. LAG(N): returns the row from the input column at the specified offset preceding
+ *      the current row. If the offset crosses the grouping boundary or column boundary for
+ *      a given row, a "default" value is returned if available. The "default" value is
+ *      null, by default.
+ *
+ * E.g. Consider an input column with the following values and grouping:
+ *      [10, 11, 12, 13,   20, 21, 22, 23]
+ *      <------G1----->   <------G2------>
+ *
+ * LEAD(input_col, 1) yields:
+ *      [11, 12, 13, null,  21, 22, 23, null]
+ *
+ * LAG(input_col, 2) yields:
+ *      [null, null, 11, 12, null, null, 21, 22]
```

Comment:
It took me a couple of times reading this to realize that input_col here is the mutated column from the LEAD call above. Should we make that more explicit with `input_col = LEAD(input_col, 1) yields:` or show the operation on the original input_col?

Reply:
As with all the rolling window functions, `offset_rolling_window()` returns a `unique_ptr<column>`. It doesn't mutate the input.

Reply:
I guess my assumption would be that `LAG(input_col, 2)` would make `[null, null, 10, 11, null, null, 20, 21]`. I'm not sure what happened to the 10 and 20 in the example unless we used the output of the above LEAD call which removed them as input for this `LAG` call.

Reply:
Oh, I see what you're pointing out. The example is incorrect. 

Thank you for the correction, @hyperbolic2346. :]

Reply:
I have corrected this example, over at #6277.